Code/Framework/AzCore/AzCore/std/typetraits/has_member_function.h

```diff
@@ -5,76 +5,141 @@
-    struct _HasName##ResultType<false, dummy> { typedef AZStd::false_type type; };                                 \
-    template<class T, bool isClass = AZStd::is_class<T>::value >                                                   \
-    class Has##_HasName {                                                                                          \
-        typedef char Yes;                                                                                          \
-        typedef long No;                                                                                           \
-        template<class U, U>                                                                                       \
-        struct TypeCheck;                                                                                          \
-        template<class U>                                                                                          \
-        struct Helper { typedef _ResultType (* mfp) _ParamsSignature; };                                           \
-        template<class C>                                                                                          \
-        static Yes ClassCheck(TypeCheck< typename Helper<C>::mfp, & C::_FunctionName>*);                           \
-        template<class C>                                                                                          \
-        static No  ClassCheck(...);                                                                                \
-    public:                                                                                                        \
-        static const bool value = (sizeof(ClassCheck< typename AZStd::remove_const<T>::type >(0)) == sizeof(Yes)); \
-        typedef typename _HasName##ResultType<value>::type type;                                                   \
-    };                                                                                                             \
-    template<class T >                                                                                             \
-    struct Has##_HasName<T, false> {                                                                               \
-        static const bool value = false;                                                                           \
-        typedef AZStd::false_type type;                                                                            \
-    };
+/**
+* AZ_HAS_MEMBER_INCLUDE_BASE detects if the class or any of its base classes
+* can invoke the member function with the specified function name and signature
+* AZ_HAS_MEMBER_INCLUDE_BASE(IsReadyIncludeBase,IsReady,void,())
+*     struct A {};
+*     struct B
+*     {
+*        static void IsReady();
+*     };
+*     struct Der : B {};
+*     HasIsReadyIncludeBase<A>::value (== false)
+*     HasIsReadyIncludeBase<B>::value (== true)
+*     HasIsReadyIncludeBase<Der>::value (== true)
```

Comment:
The name in the example doesn't match the one defined above (`IsReadyIncludeBase` vs. `HasIsReadyIncludeBase`).

Reply:
Repeat: The term "Has" is always prepended to the first argument of AZ_HAS_MEMBER_INCLUDE_BASE